Check whether path matches the given pattern prior to linting (issue #16)

patbuergin opened the pull request.

es128 commented on src/index.coffee:
```diff
@@ -48,39 +48,44 @@ module.exports = class JSHintLinter
-        results = errors.map (error) ->
-          error: error
-          file: path
+        if @reporter
```
Why the extra level of indentation for everything from here down?

Since the `if` above (line 57) ends with a short-circuit `return`, what should actually happen is eliminating the `else` - just do the `errors` filter operation and proceed with the rest of this at the same indent as before.

es128 commented:
Thank you for contributing. 

This patch causes two tests to fail currently. If you can resolve that and the indentation issue, should be good to merge and publish this.

patbuergin commented:
I've fixed the broken tests and added two new ones to cover the previously disregarded `pattern` config field. The superfluous `else` statements increase readability in my opinion (`if success [...] else`), but I agree on the indentation concern and reduced the level accordingly.
